swagger/swagger-generator/generator-core/src/main/java/io/servicecomb/swagger/generator/core/utils/ClassUtils.java

```diff
@@ -217,6 +275,20 @@ public static String correctMethodParameterName(String name) {
     }
 
     public static String correctClassName(String name) {
-        return name.replace("-", "_");
+        String parts[] = name.split("\\.");
+        for (int idx = 0; idx < parts.length; idx++) {
+            String part = parts[idx];
+            if (part.isEmpty()) {
+                parts[idx] = "_";
```

Comment:
at first, i don't think anyone will use a name like "1..."
  and add "part.isEmpty()" is because "part.charAt(0)" in the next block.

but, ok, i will add test case for this.